### ISSUE TITLE
Support for TOML in config files?

### ISSUE THREAD

**Issue by ChristianSi** [zerobuzz/configifier#2]
Personally I don't like YAML very much, as it is overly complicated. Maybe it would make sense to add support for [TOML](https://github.com/toml-lang/toml) which is a much more lightweight format?

Downside is that Haskell currently lacks an up-to-date and well-maintained TOML parser.

**Comment by mf59816** [comment]
Good idea.  (Note though that the `yaml` package also reads json, but toml probably has advantages over both.)

I would implement this thusly:
1. find a good-enough toml parser lib.  (or write one.)
2. extend `data Source` with another constructor `ConfigFileToml`.
3. add a pattern in `configify` that translates toml into `Aeson.Value`, and then call `parseConfigFile`

I will do this as soon as I run out of ideas that I consider more important (currently: many (sorry :)).  Pull requests are always welcome, of course
